services/motion/motion.go

```diff
@@ -40,6 +40,19 @@ type PlanHistoryReq struct {
 	Extra       map[string]interface{}
 }
 
+// MoveReq describes the request to the Move interface method.
+type MoveReq struct {
+	// ComponentName of the component to move
+	ComponentName resource.Name
+	// Goal destination the component should be moved to
+	Destination referenceframe.PoseInFrame
+	// The external environment to be considered for the duration of the move
+	WorldState referenceframe.WorldState
+	// Constraints which need to be satisfied during the movement
+	Constraints motionplan.Constraints
```

Comment:
Previously we were accepting pointers for all of the above, but this gets a little weird when you put it into a struct since the default if unset becomes a nil.  Technically keeping them as pointers wouldnt be a regression because it was always possible to pass nils into the function, so I could see the argument for minimizing change by going back to making them pointers.

Reply:
IMO pointers are better.

Imagine the case where a Destination is not provided. If this is a pointer, then as you say, it is nil; easy to check. If it is a PoseInFrame, then an uninitialized Destination is a real PoseInFrame with a nil underlying Pose, and an empty string for parent frame. This is much more difficult to check for, and for some structs may have overlap with actual valid, meaningful inputs.

Reply:
Cool I'll revert back to this then. I think it will make the code easier to read too.  I take it based on your lack of other comments that you're cool with the structifying and the general direction of the PR?

Reply:
As far as it pertains to `Move` yes.

I think we'll want the DoCommand API to be much more fully featured than what you have wireframed.

Reply:
Yeah I agree.  Do you have ideas of what you want to see in it now?  I will do my best to include

Reply:
We will need things like being able to request motion from `motionplan` whose goal is a joint configuration, rather than a pose.

We will need to be able to request a motion that starts from an arbitrary set of joint positions, rather than the current ones.

Reply:
Oh right thats a good callout thanks